pkg/dio/multi.go

```diff
@@ -192,6 +192,11 @@ func (t *multiReader) Read(p []byte) (int, error) {
 		}
 	}
 
+	// return 0, nil if there are no readers
+	if len(t.readers) == 0 {
```

Comment:
yes, but @caglar10ur was going to alter this to use a tmpvar anyway, as with the Writer impl.